quicksort/quicksort.c

```diff
@@ -17,7 +17,27 @@
 */
 void quicksort(int *arr, int low, int high)
 {
-
+    int again = 1;
+    int count = low;
+    int noswap = 0;
+    while (again == 1){
+        if (count == high ){
+            count = low;
+            if ((high-low) == noswap){
+                again = 0;
+            }
+            else{
+                noswap = 0;
+            }
+        }
+        if (arr[count] > arr[count + 1]){
+            swap(&arr[count], &arr[count +1]);
+        }
+        else{
+            noswap +=1;
+        }
+        count += 1;
+    }
 }
```

Comment:
This does look more like a bubble sort than a quicksort.
https://www.geeksforgeeks.org/quick-sort/ -> this quicksort solution involves creating a helper function and then running quicksort recursively, just for future reference.